### PR TITLE
[REVIEW] Add bzip2 decompression support to read_csv()

### PR DESCRIPTION
**Pull request by OlivierNV**
Add bzip2 decompression support to read_csv()
Fixes issue #701

### CODE REVIEW

File: cpp/src/io/comp/uncomp.cpp [hunk 1]
```diff
@@ -372,7 +392,37 @@ gdf_error io_uncompress_single_h2d(const void *src, gdf_size_type src_size, int
...
+            dst.resize(0);
+            return GDF_FILE_ERROR;
+        }
+    }
+    else if (strm_type == IO_UNCOMP_STREAM_TYPE_BZIP2)
+    {
+        size_t src_ofs = 0;
+        size_t dst_ofs = 0;
+        int bz_err = 0;
+    	dst.resize(uncomp_len);
```
Comment (j-ieong) [inline cpp/src/io/comp/uncomp.cpp]: Missing one space for alignment here

Comment (OlivierNV) [reply cpp/src/io/comp/uncomp.cpp]: I don't seem to see it in the source, it looks like the problem is there are four spaces followed by a tab (copied from line 390 which also includes 4 spaces & 1 tab). Will replace both with spaces.

Comment (OlivierNV) [reply cpp/src/io/comp/uncomp.cpp]: Added a commit to fix indentation

**Comment by harrism** [comment]
rerun tests

**Comment by mjsamoht** [comment]
@OlivierNV can you rerun the checks?  TOT build should be fixed now.  After that we can merge your PR.

**Comment by OlivierNV** [comment]
rerun tests

File: cpp/src/io/comp/uncomp.cpp [hunk 2]
```diff
@@ -350,12 +356,27 @@ gdf_error io_uncompress_single_h2d(const void *src, gdf_size_type src_size, int
...
             }
-            if (strm_type != IO_UNCOMP_STREAM_TYPE_INFER)
-                break; // Fall through for INFER
-        default:
-            // Unsupported format
-            break;
         }
+        if (strm_type != IO_UNCOMP_STREAM_TYPE_INFER)
+            break; // Fall through for INFER
+    case IO_UNCOMP_STREAM_TYPE_BZIP2:
```
Comment (harrism) [inline cpp/src/io/comp/uncomp.cpp]: Indentation appears to be inconsistent below here.

File: cpp/src/io/comp/uncomp.cpp [hunk 3]
```diff
@@ -350,12 +356,27 @@ gdf_error io_uncompress_single_h2d(const void *src, gdf_size_type src_size, int
...
+        if (src_size > 4)
+        {
+            const bz2_file_header_s *fhdr = (const bz2_file_header_s *)raw;
+            // Check for BZIP2 file signature "BZh1" to "BZh9"
+            if (fhdr->sig[0] == 'B' && fhdr->sig[1] == 'Z' && fhdr->sig[2] == 'h' && fhdr->sig[3] >= '1' && fhdr->sig[3] <= '9')
```
Comment (harrism) [inline cpp/src/io/comp/uncomp.cpp]: For the `BZh` part, why not std::string::compare() or strncmp for clarity/brevity?

**Comment by OlivierNV** [comment]
rerun tests

**Comment by mjsamoht** [comment]
@OlivierNV the fix for the style check failures are merged.  Can you rerun your checks?

**Comment by kkraus14** [comment]
rerun tests

File: cpp/src/io/comp/uncomp.cpp (hunk 3 above)
Comment (OlivierNV) [reply cpp/src/io/comp/uncomp.cpp]: I personally prefer individual checks for 3-characters on something that is not a zero-terminated string (Ideally would have liked a 32-bit word signature with if (sig >= 'BZh1' && sig <= 'BZh9') but some compilers complain that multi-character constants are non-standard (presumably due to implied assumption about endianness).

**Pull request merged**